velox/exec/tests/SortBufferTest.cpp

```diff
@@ -0,0 +1,163 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/exec/SortBuffer.h"
+#include <gtest/gtest.h>
+#include <type/Type.h>
+#include <filesystem>
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+using namespace facebook::velox::exec;
+using namespace facebook::velox;
+using namespace facebook::velox::memory;
+
+namespace facebook::velox::functions::test {
+
+class SortBufferTest : public testing::Test,
+                       public velox::test::VectorTestBase {
+ public:
+  // c0,         c1,         c2,         c3,         c4,         c5
+  // BIGINT()    INTEGER()   SMALLINT()  REAL()      DOUBLE()    VARCHAR()
+  //             sorted-2                            sorted-1
+  RowTypePtr inputType_ = ROW(
+      {{"c0", BIGINT()},
+       {"c1", INTEGER()},
+       {"c2", SMALLINT()},
+       {"c3", REAL()},
+       {"c4", DOUBLE()},
+       {"c5", VARCHAR()}});
+  std::vector<column_index_t> sortColumnIndices_{4, 1};
+  std::vector<CompareFlags> sortCompareFlags_{
+      {true, true, false, false},
+      {true, true, false, false}};
+
+  std::shared_ptr<memory::MemoryPool> rootPool_{
+      memory::defaultMemoryManager().addRootPool("SortBufferTest")};
+  std::shared_ptr<memory::MemoryPool> pool_{
+      rootPool_->addLeafChild("SortBufferTest")};
+  tsan_atomic<bool> nonReclaimableSection_{false};
+};
+
+TEST_F(SortBufferTest, basic) {
+  auto sortBuffer = new SortBuffer(
+      inputType_,
+      sortColumnIndices_,
+      sortCompareFlags_,
+      pool_.get(),
+      &nonReclaimableSection_);
+
+  RowVectorPtr data = makeRowVector(
+      {makeFlatVector<int64_t>({1, 2, 3, 4, 5}),
+       makeFlatVector<int32_t>({5, 4, 3, 2, 1}), // sorted-2 column
+       makeFlatVector<int16_t>({1, 2, 3, 4, 5}),
+       makeFlatVector<float>({1.1, 2.2, 3.3, 4.4, 5.5}),
+       makeFlatVector<double>({1.1, 2.2, 2.2, 5.5, 5.5}), // sorted-1 column
+       makeFlatVector<std::string>(
+           {"hello", "world", "today", "is", "great"})});
+
+  sortBuffer->addInput(data);
+  sortBuffer->noMoreInput();
+  auto output = sortBuffer->getOutput();
+  ASSERT_EQ(output->size(), 5);
+  ASSERT_EQ(output->childAt(1)->asFlatVector<int32_t>()->valueAt(0), 5);
+  ASSERT_EQ(output->childAt(1)->asFlatVector<int32_t>()->valueAt(1), 3);
+  ASSERT_EQ(output->childAt(1)->asFlatVector<int32_t>()->valueAt(2), 4);
+  ASSERT_EQ(output->childAt(1)->asFlatVector<int32_t>()->valueAt(3), 1);
+  ASSERT_EQ(output->childAt(1)->asFlatVector<int32_t>()->valueAt(4), 2);
+}
+
+TEST_F(SortBufferTest, batchOutput) {
+  auto sortBuffer = new SortBuffer(
+      inputType_,
+      sortColumnIndices_,
+      sortCompareFlags_,
+      pool_.get(),
+      &nonReclaimableSection_);
```

Comment:
I think it might be better to structure the test as
```
struct {
   bool triggerSpill;
   size_t numInputRows;
   size_t outputBatchSize; -> test 1 such special case. Also please add CHECK in sort buffer ctor, outputBatchSize can't be zero > 0
  std::vector<size_t> expectedOutputBufferSizes;
} testSettings[] = { => list different test scenarios and check existing test cases for example
```